Bottom: 8da1ee3208c85966521e6df3ebcf748e4bbc2352
Top:    49368c9046bbf6c56a38db0d7eaf60726191e15b
Author: Alexandre Oliva <aoliva@redhat.com>
Date:   2016-04-07 03:05:05 -0300

make the user expression function a global friend

---

diff --git a/libcc1/libcp1plugin.cc b/libcc1/libcp1plugin.cc
index 3808017..281b7ca 100644
--- a/libcc1/libcp1plugin.cc
+++ b/libcc1/libcp1plugin.cc
@@ -262,6 +262,11 @@ static void
 plugin_pragma_user_expression (cpp_reader *)
 {
   cp_binding_oracle = plugin_binding_oracle;
+
+  /* Make the function containing the user expression a global
+     friend, so as to bypass access controls in it.  */
+  if (at_function_scope_p ())
+    add_to_global_friend_list (current_function_decl);
 }
 
 static void
